docs/paper/dev/api/event-api/chat-event.md

```diff
@@ -0,0 +1,156 @@
+- The `sourceDisplayName` parameter is the display name of the player that sent the message.
+- The `message` parameter is the message that was sent.
+- The `viewer` parameter is the player that is receiving the message.
+
+Now that we understand how the renderer works, we can start using it.
+
+## Using the Renderer
+
+There are two ways to use the renderer.
+1. Creating a new Class that implements `ChatRenderer`. 
+2. Using a lambda expression.
+
+Depending on the complexity of your renderer you may want to use one or the other.
+
+### Creating a new Class
+
+If you want to use a class for your renderer, you just need to implement the `ChatRenderer` interface.
+
+```java title="ExampleRenderer.java"
+public class ExampleRenderer implements ChatRenderer {
+
+    @Override
+    public @NotNull Component render(@NotNull Player source, @NotNull Component sourceDisplayName, @NotNull Component message, @NotNull Audience viewer) {
+        // ...
+    }
+
+}
+```
+
+To be able to use this renderer, we need to tell the event to use it.
+
+```java title="ChatEvent.java"
+public class ChatEvent implements Listener {
+
+    @EventHandler
+    public void onChat(AsyncChatEvent event) {
+        event.renderer(new ExampleRenderer());
+    }
+
+}
+```
+
+### Using a lambda expression
+
+Another way of using the renderer is by using a lambda expression.
+
+```java title="ChatEvent.java"
+public class ChatEvent implements Listener {
+
+    @EventHandler
+    public void onChat(AsyncChatEvent event) {
+        event.renderer((source, sourceDisplayName, message, viewer) -> {
+            // ...
+        });
+    }
+
+}
+```
+
+## Rendering the message
+
+Now that we have our renderer, we can start rendering the message.
+
+Let's say we want to render our chat to look like this:
+
+![](./assets/plain-message-rendering.png)
+
+To do this, we need to return a new `Component` that contains the message we want to send.
+
+```java title="ExampleRenderer.java"
+public class ExampleRenderer implements ChatRenderer {
+
+    @Override
+    public @NotNull Component render(@NotNull Player source, @NotNull Component sourceDisplayName, @NotNull Component message, @NotNull Audience viewer) {
+        return sourceDisplayName 
+                .append(Component.text(": ")) 
+                .append(message); 
+    }
+
+}
+```
+
+Now you can see that the message is rendered as we wanted it to be.
+
+## Rendering the message with a prefix using LuckPerms and MiniMessage
+
+Usually, you would want to add a prefix to the message. 
+In this example, we will be using LuckPerms to get the players prefix.
+Let's say we want to render our chat to look like this:
+
+![img.png](assets/prefix-rendering.png)
+
+This process is a bit more complex than the previous one.
+For this we create our `admin` group and add a prefix to it:
+
+- /lp creategroup admin
+
+- /lp group admin meta addprefix 100 "`<gray>[</gray><red>Admin</red><gray>]</gray>`"
```

Comment:
Why is the prefix in backticks here?

Reply:
without backticks it produces this: 
![image](https://github.com/PaperMC/docs/assets/59799222/82865e2b-3692-465f-80e8-22c4f2702a92)

Reply:
Put the whole command in backticks?